lib/addons/core-addon/index.ts

```diff
@@ -43,27 +47,44 @@ export class CoreAddOn implements ClusterAddOn {
         // Create a service account if user provides namespace and service account
         let serviceAccountRoleArn: string | undefined = undefined;
         let serviceAccount: ServiceAccount | undefined = undefined;
+        let saNamespace: string | undefined = undefined;
+
+        saNamespace = DEFAULT_NAMESPACE;
+        if (this.coreAddOnProps?.namespace) {
+            saNamespace = this.coreAddOnProps.namespace;
+        }
 
         if (this.coreAddOnProps?.policyDocumentProvider) {
             const policyDoc = this.coreAddOnProps.policyDocumentProvider(clusterInfo.cluster.stack.partition);
             serviceAccount  = createServiceAccount(clusterInfo.cluster, this.coreAddOnProps.saName,
-                DEFAULT_NAMESPACE, policyDoc);
+                saNamespace, policyDoc);
             serviceAccountRoleArn = serviceAccount.role.roleArn;
 
         }
-        const cfnaddon = new CfnAddon(clusterInfo.cluster.stack, this.coreAddOnProps.addOnName + "-addOn", {
-            addonName: this.coreAddOnProps.addOnName,
-            addonVersion: this.coreAddOnProps.version,
-            clusterName: clusterInfo.cluster.clusterName,
-            serviceAccountRoleArn: serviceAccountRoleArn,
-            resolveConflicts: "OVERWRITE"
-        });
-        
-        if (serviceAccount) {
-            cfnaddon.node.addDependency(serviceAccount);
+        if (this.coreAddOnProps.version) {
+            const cfnaddon = new CfnAddon(clusterInfo.cluster.stack, this.coreAddOnProps.addOnName + "-addOn", {
```

Comment:
let's fix the code duplication on line 65-73 qne 78 - 87 as we discussed.

Reply:
This is addressed with never changes to CoreAddon to fix all issues. Please check.